src/main/java/com/adorsys/keycloakstatuslist/service/RevocationRecordService.java

```diff
@@ -38,77 +95,37 @@ public TokenStatusRecord createRevocationRecord(CredentialRevocationRequest requ
         
         try {
             RealmModel realm = session.getContext().getRealm();
-            
             validateRevocationReason(request.getRevocationReason());
             
-            String[] keyAndAlg = getRealmPublicKeyAndAlg(realm);
-            String publicKey = keyAndAlg[0];
-            String algorithm = keyAndAlg[1];
+            KeyData keyData = getRealmKeyData(session, realm);
             
             TokenStatusRecord record = new TokenStatusRecord();
             record.setCredentialId(request.getCredentialId());
             record.setIssuer(realm.getName());
             record.setIssuerId(realm.getName());
-            record.setPublicKey(publicKey);
-            record.setAlg(algorithm);
+            
+            record.setPublicKey(keyData.jwk());
+            record.setAlg(keyData.algorithm());
+            
             record.setStatus(TokenStatus.REVOKED);
             record.setCredentialType("oauth2");
             record.setRevokedAt(Instant.now());
             record.setStatusReason(request.getRevocationReason() != null && !request.getRevocationReason().trim().isEmpty() ? 
                                  request.getRevocationReason() : "Credential revoked");
             
-            logger.infof("Created revocation record. RequestId: %s, CredentialId: %s, Status: %s", 
-                         requestId, record.getCredentialId(), record.getStatus());
-            
-            logger.infof("Revocation record details - Issuer: %s, Algorithm: %s, Reason: %s", 
-                         record.getIssuer(), record.getAlg(), record.getStatusReason());
-            
             return record;
             
         } catch (Exception e) {
             logger.errorf("Failed to create revocation record. RequestId: %s, Error: %s", 
                          requestId, e.getMessage());
+            if (e instanceof StatusListException) throw (StatusListException) e;
             throw new StatusListException("Failed to create revocation record: " + e.getMessage(), e);
```

Comment:
There is no issue wrapping a StatusListException inside another one.
```suggestion
throw new StatusListException("Failed to create revocation record: " + e.getMessage(), e);
```